src/asymmetric_crypto/curve_25519/x25519/key_pair.rs

```diff
@@ -0,0 +1,166 @@
+                    X25519PrivateKey::LENGTH,
+                    None,
+                )
+            })?,
+        )?;
+        let public_key = X25519PublicKey::try_from_bytes(
+            bytes[X25519PrivateKey::LENGTH..].try_into().map_err(|_| {
+                crate::CryptoCoreError::InvalidBytesLength(
+                    "key pair (public key)".to_string(),
+                    X25519PrivateKey::LENGTH,
+                    None,
+                )
+            })?,
+        )?;
+        Ok(Self {
+            private_key,
+            public_key,
+        })
+    }
+}
+
+impl Zeroize for X25519Keypair {
+    fn zeroize(&mut self) {
+        self.private_key.zeroize();
+    }
+}
+
+impl Drop for X25519Keypair {
+    fn drop(&mut self) {
+        self.zeroize();
+    }
+}
+
+impl ZeroizeOnDrop for X25519Keypair {}
+
+impl X25519Keypair {
+    pub fn new<R: CryptoRngCore>(rng: &mut R) -> Result<Self, CryptoCoreError> {
+        let private_key = X25519PrivateKey::new(rng);
+        let public_key = X25519PublicKey::from(&private_key);
+        Ok(Self {
+            private_key,
+            public_key,
+        })
+    }
+}
+
+/// Algorithm [`ObjectIdentifier`] for the X25519 digital signature algorithm
+/// (`id-X25519`).
+///
+/// <http://oid-info.com/get/1.3.101.110>
+pub const X25519_ALGORITHM_OID: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.3.101.110");
+
+/// X25519 Algorithm Identifier.
+pub const X25519_ALGORITHM_ID: pkcs8::AlgorithmIdentifierRef<'static> =
+    pkcs8::AlgorithmIdentifierRef {
+        oid: X25519_ALGORITHM_OID,
+        parameters: None,
+    };
+
+impl EncodePublicKey for X25519Keypair {
+    fn to_public_key_der(&self) -> spki::Result<Document> {
+        pkcs8::SubjectPublicKeyInfoRef {
+            algorithm: X25519_ALGORITHM_ID,
+            subject_public_key: BitStringRef::new(0, self.public_key.as_bytes())?,
+        }
+        .try_into()
+    }
+}
+
+impl EncodePrivateKey for X25519Keypair {
+    fn to_pkcs8_der(&self) -> pkcs8::Result<pkcs8::SecretDocument> {
+        // Serialize private key as nested OCTET STRING
+        let mut private_key = [0u8; 2 + (Self::LENGTH / 2)];
+        private_key[0] = 0x04;
+        private_key[1] = 0x20;
+        private_key[2..].copy_from_slice(self.private_key.as_bytes());
+
+        let private_key_info = PrivateKeyInfo {
+            algorithm: X25519_ALGORITHM_ID,
+            private_key: &private_key,
+            public_key: None, // public_key: Some(self.public_key.as_bytes()),
+        };
+
+        let result = SecretDocument::encode_msg(&private_key_info)?;
+
+        Ok(result)
+    }
+}
+
+impl SignatureAlgorithmIdentifier for X25519Keypair {
+    type Params = AnyRef<'static>;
+
+    const SIGNATURE_ALGORITHM_IDENTIFIER: AlgorithmIdentifier<Self::Params> = X25519_ALGORITHM_ID;
+}
+
+pub struct X25519Signature {
+    pub(crate) signature: ed25519_dalek::Signature,
+}
+impl X25519Signature {
```

Comment:
It is strange that your clippy corrects missing trailing `;` but allows this.

Reply:
file finally removed as it appears we won't need this impl